daemon/SecretsImpl/secrets.cpp

```diff
@@ -747,12 +746,20 @@ QByteArray Daemon::ApiImpl::SecretsRequestQueue::saltData() const
         return QByteArray();
     }
 
-    const QString saltFileName = m_autotestMode
+    const QString saltDirName = m_autotestMode
             ? QLatin1String("initialsalt-test")
             : QLatin1String("initialsalt");
-    const QString saltPath = secretsDir.absoluteFilePath(saltFileName);
-    if (!QFile::exists(saltPath)) {
-        // first run, need to write the initial salt data file.
+    const QString saltDirPath = secretsDir.absoluteFilePath(saltDirName);
+
+    DataProtector dataProtector(saltDirPath);
+    QByteArray saltData;
+    DataProtector::Status s = dataProtector.getData(&saltData);
+    if (s != DataProtector::Success) {
+        qWarning(lcSailfishSecretsDaemon) << "Salt data is corrupted.";
```

Comment:
Again wording make this distinguishable from line 778.

Reply:
When initialsalt is a file e.i. one is upgrading secrets => it's always "Salt data is corrupted". Making in practice previous secrets void. Maybe we can live with this as we haven't never related this before.

Reply:
Well, do you want to support this or not? If yes, then there is a commit that deals with it. If not, then just accept that you have to delete the old data when you upgrade.